### PR TITLE
Enhancement/output config

### PR DESCRIPTION
**Pull request by mmcleod89**

### CODE REVIEW

File: src/hivpy/cli.py [hunk 1]
```diff
@@ -18,21 +19,24 @@
...
+    parser.add_argument("input", type=pathlib.Path, help="run_model config.yaml")
     args = parser.parse_args()
-    conf_filename = args.input
-    config = configparser.ConfigParser()
+    config_filename = args.input
+    try:
+        with open(config_filename, 'r') as file:
+            config = yaml.safe_load(file)
+    except yaml.YAMLError as err:
+        print("Error parsing yaml file {}".format(err))
```
Comment (ageorgou) [inline src/hivpy/cli.py:33]: Maybe we can print the error in a more readable way, but probably fine to leave this until we get actual errors and feedback from users.

File: src/hivpy/simulation.py [hunk 2]
```diff
@@ -82,38 +79,28 @@ def update_summary_stats(self, date, pop_data):
...
     population: Population
-    results: pd.DataFrame
 
     def __init__(self, simulation_config):
         self.simulation_config = simulation_config
-        self.results = None
         self._initialise_population()
         self.output = SimulationOutput(self.simulation_config)
+        self.output_path = simulation_config.output_dir / (
+            "simulation_output." + str(datetime.now().strftime("%Y%m%d-%H%M%S")) + ".csv")
```
Comment (ageorgou) [inline src/hivpy/simulation.py]: Unrelated to these changes, but it might be better to have the filename as `simulation_output_{date}.csv` instead, to avoid the extra `.` which can be confusing/mess up how the name is displayed?

File: src/tests/test_experiment.py [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
     """Sample experiment parameters for testing."""
-    parser = ConfigParser()
-    parser.read(
-        os.path.join(os.path.dirname(__file__), 'fixtures', 'sample.conf'))
-    print(list(parser.items()))
-    return parser
+    filepath = os.path.join(os.path.dirname(__file__), 'fixtures', 'sample.yaml')
+    with open(filepath, 'r') as sample_file:
+        sample_config = yaml.safe_load(sample_file)
+    print(sample_config)
```
Comment (ageorgou) [inline src/tests/test_experiment.py]: Oops, this was probably left in from debugging the config parser in the past!

File: src/tests/test_simulation.py [hunk 4]
```diff
@@ -43,40 +44,21 @@ def test_error_end_before_first_step(tmp_path):
...
-    pop, results = run_simulation(config)
+    config = SimulationConfig(size, start, end, tmp_path, step)
+    simulation_handler = SimulationHandler(config)
+    simulation_handler.run()
+    pop = simulation_handler.population
     # Check that the number alive never grows... (some steps may have 0 deaths)
-    assert all(results.num_alive.diff()[1:] <= 0)
+    # assert all(results.num_alive.diff()[1:] <= 0)
+    num_not_dead = sum(pop.data.date_of_death.isnull())
+    assert num_not_dead < len(pop.data)
```
Comment (ageorgou) [inline src/tests/test_simulation.py:61]: Can we check this from the SimulationOutput instead? (If we don't store the number of people alive there, we should! though that can be a different PR)

File: hivpy.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+EXPERIMENT: 
+  population: 1000
+  start_year: 1989
+  end_year: 2021
+  time_interval_days: 90
+  simulation_output_dir: output
```
Comment (ageorgou) [inline hivpy.yaml:6]: Looking back at [the issue comment](https://github.com/UCL/hivpy/issues/31#issuecomment-1131793090), I'd noted that we could move this out of this section and instead have a general "Output" section with the logging and other output options. I don't see any problem doing it this way either (probably better for consistency too), just writing it down so there's a record if we look back in the future!

**Review by ageorgou — APPROVED** [review]
All looks good, just some small comments. We should also delete the `hivpy.example.conf` file at the top level.

File: src/hivpy/simulation.py (hunk 2 above)
Comment (mmcleod89) [reply src/hivpy/simulation.py]: It shouldn't cause any issues for the name being displayed (it doesn't on my Windows machine at least) but I don't mind changing the style (in which case we should change the style of the log output files as well which are done similarly).

File: src/tests/test_experiment.py (hunk 3 above)
Comment (mmcleod89) [reply src/tests/test_experiment.py]: Okay I will remove it!

File: src/tests/test_simulation.py (hunk 4 above)
Comment (mmcleod89) [reply src/tests/test_simulation.py:61]: I think we can add in something that checks the simulation output too, but I'm inclined to also check that the population object is itself updating correctly as that's the thing that's going to be accessed by other parts of the simulation that are affected by this.
